### PR TITLE
Added ggus_alerting files, updated ssb_alerting files

### PR DESCRIPTION
**Pull request by indrarahul2013**
Implementation of alerting tool for the GGUS Ticketing System which will ingest JSON data and inject them into AlertManager. Ending Time for each alerts has been implemented on the logic of checking if a particular ticket has been resolved or not, and updating the same in AlertManager.

### CODE REVIEW

File: src/go/MONIT/ggus_alerting.go [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
```
Comment (vkuznet) [inline src/go/MONIT/ggus_alerting.go:215]: Will it work for alerts which will have different attributes then ggus ones?

File: src/go/MONIT/ggus_alerting.go [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+		Subject         string `json:"Subject"`
+		Scope           string `json:"Scope"`
+		URL             string `json:"URL"`
+	} `json:"annotations"`
+	StartsAt time.Time `json:"startsAt"`
+	EndsAt   time.Time `json:"endsAt"`
+}
+
+//AlertManager GET API acceptable JSON Data struct for GGUS data
+type getAMJSON struct {
```
Comment (vkuznet) [inline src/go/MONIT/ggus_alerting.go:75]: What is the reason to name a struct with *get* prefix? The get prefix indicates that you get the data, so it is more appropriate for a function name. Please rename it appropriately, e.g. ggusData.

File: src/go/MONIT/ggus_alerting.go [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+	ResponsibleUnit string `json:"ResponsibleUnit"`
+	Status          string `json:"Status"`
+	LastUpdate      int    `json:"LastUpdate"`
+	Subject         string `json:"Subject"`
+	Scope           string `json:"Scope"`
+}
+
+//AlertManager API acceptable JSON Data for GGUS Data
+type amJSON struct {
+	Labels struct {
```
Comment (vkuznet) [inline src/go/MONIT/ggus_alerting.go:53]: This part repeats in getAMJSON struc, therefore take it out and re-use ,e.g.
```
type Labels struct {
  ...
}
type amJSON struct {
....
   Labels Labels
...
}
```

File: src/go/MONIT/ggus_alerting.go [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+		Alertname string `json:"alertname"`
+		Severity  string `json:"severity"`
+		Tag       string `json:"tag"`
+	} `json:"labels"`
+	Annotations struct {
```
Comment (vkuznet) [inline src/go/MONIT/ggus_alerting.go:58]: Similar here, the Annotations is identical in both structs, in amJSON and in getAMJSON. You should define it outside of them and then use it.

File: src/go/MONIT/ggus_alerting.go [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+
+	var temp amJSON
+	var finalData []amJSON
+
+	exstTkt = make(map[string]int)
+
+	for _, each := range data {
+
+		// Ignoring all tickets whose VO is not "cms"
+		if each.VO != "cms" {
```
Comment (vkuznet) [inline src/go/MONIT/ggus_alerting.go:149]: Make it configurable, such that if we will need to change it later we'll have this flexibility, i.e. we should be able to pass filter to the program, but it should have default value "cms".

File: src/go/MONIT/ggus_alerting.go (hunk 1 above)
Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go:215]: Yes ! I have injected both alerts (SSB & GGUS Data) and it works independently. Please let me know if I still have to filter the GGUS data.

File: src/go/MONIT/ggus_alerting.go (hunk 5 above)
Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go:149]: Fixed this issue.

File: src/go/MONIT/ggus_alerting.go (hunk 4 above)
Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go:61]: Fixed the getAMJSON struct. Named to ggusData, removed unused variables.

File: src/go/MONIT/ggus_alerting.go (hunk 3 above)
Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go:56]: Fixed this.

File: src/go/MONIT/ggus_alerting.go (hunk 2 above)
Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go]: Renamed it to ggusData.

File: src/go/MONIT/ggus_alerting.go [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+		VO              string `json:"VO"`
+		URL             string `json:"URL"`
+	} `json:"annotations"`
+	StartsAt time.Time `json:"startsAt"`
+	EndsAt   time.Time `json:"endsAt"`
+}
+
+//AlertManager GET API acceptable JSON Data struct for GGUS data
+type ggusData struct {
+	Data []amJSON
```
Comment (vkuznet) [inline src/go/MONIT/ggus_alerting.go:79]: If ggusData is really a list of amJSON then why do you need this struct at all? You can easily Unmarshal data using `[]amJSON`.

Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go:79]: Valentin,
When I call /api/v1/alerts (GET) API, I get struct of data array from AlertManager. It is of form :- 
```
{
   "data" : [ {},
              {},
             ],
. . . 
}
```

So, I would be requiring struct to parse the information.

Comment (vkuznet) [reply src/go/MONIT/ggus_alerting.go:79]: ok, then it is correct.

File: src/go/MONIT/ggus_alerting.go (hunk 1 above)
Comment (indrarahul2013) [reply src/go/MONIT/ggus_alerting.go:189]: I have added the filter for fetching only GGUS alerts from AlertManager. 
```
apiurl := alertManagerURL + "/api/v1/alerts?active=true&silenced=false&inhibited=false&unprocessed=false&filter=tag=\"GGUS\""
```

**Pull request merged**